src/output.c

```diff
@@ -62,16 +111,18 @@ void output_advance(output_t *st)
 
         for (frame = 0; frame < audio_frames; frame++)
         {
-            unsigned int len = elastic->packets[elastic->audio_offset].size;
-            uint8_t *pkt = elastic->packets[elastic->audio_offset].data;
+            packet_t* pkt = &elastic->packets[elastic->audio_offset];
 #ifdef USE_FAAD2
             int produced_audio = 0;
 #endif
 
-            if (len > 0)
+            if (is_complete_pkt(pkt))
             {
-                nrsc5_report_hdc(st->radio, program, pkt, len);
+                nrsc5_report_hdc(st->radio, program, pkt);
+            }
```

Comment:
Why do we skip reporting incomplete packets here?

Reply:
Wait, maybe I misunderstood something. I noticed a lot more HDC packets getting reported for my low-SNR samples, and assumed we're now intentionally reporting empty HDC packets. But maybe that's not intentional?

Reply:
Never mind. After thinking some more I realized that the "extra" HDC packets must be the ones with bad CRCs, which were not previously reported.

Reply:
It may be worth considering whether we should report "empty" HDC packets in the cases where we return silence in the audio stream.

Reply:
Oh, I think understand the confusion. I wanted empty HDC packets with CRC flag set to be reported. Is that not what is happening? 

I am not sure if you have another idea. I thought it would make sense to report HDC packets or empty HDC packets with CRC flag only if they were complete packets.

Reply:
I do think that reporting CRC errors in HDC events makes sense, and it's the same approach I was planning to take before you took care of it.

I think ideally the `NRSC5_EVENT_HDC` and `NRSC5_EVENT_AUDIO` streams should line up packet-for-packet so that the audio stream could be exactly recovered from the HDC stream if desired. But I don't think we're quite there yet, since there are circumstances where we skip reporting an HDC event.

We could leave that to a future PR if you like.

Reply:
Ohhhhh, okay. I understand now. I think that should be in a separate pr. I can work on that after this. It'll be able to discuss with you my thoughts on that.